### PR TITLE
Automate intake QAs

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/models/health/careplan_saver.rb [hunk 1]
```diff
@@ -22,57 +20,86 @@ def create
...
-              @qualifying_activity.reached_client = :yes
-            else
-              # default to email signature
-              @qualifying_activity.mode_of_contact = :other
-              @qualifying_activity.reached_client = :collateral
-              @qualifying_activity.mode_of_contact_other = 'On-line'
-              @qualifying_activity.reached_client_collateral_contact = 'On-line Signature'
-            end
+          if @careplan.just_finished? && @create_qa
+            qualifying_activity = setup_care_planning_qualifying_activity
```
Comment (eanders) [inline app/models/health/careplan_saver.rb]: If both signatures are collected in one go (added manually, for instance) do we need to generate both QA?  I believe you'll only get the provider as written.

File: app/models/health/careplan_saver.rb [hunk 2]
```diff
@@ -22,57 +20,86 @@ def create
...
+      )
+    end
+
+    protected def setup_pctp_signed_qualifying_activity
+      case @careplan.provider_signature_mode.to_s
+      when 'email'
+        mode_of_contact = :other
+        reached_client = :collateral
+        mode_of_contact_other = 'On-line'
+        reached_client_collateral_contact = 'On-line Signature'
```
Comment (eanders) [inline app/models/health/careplan_saver.rb:82]: If the options are 'in_person' and 'everything else' seems like we probably don't need the email case.

File: app/models/health/cha_saver.rb [hunk 3]
```diff
@@ -37,31 +33,30 @@ def create
...
 
-
     def update
       @cha.class.transaction do
         @cha.completed_at = nil unless @complete
         @cha.save!
-        if @complete && @reviewed && @create_qa
-          @qualifying_activity.source_id = @cha.id
-          @qualifying_activity.save
+        if @complete && @reviewed && @create_qa && @cha.patient.recent_ssm_form.completed_at.present?
```
Comment (eanders) [inline app/models/health/cha_saver.rb:42]: I'm confused, why is the CHA checking for a recent completed SSM?  Maybe a comment as to the intention is necessary.

File: app/models/health/cha_saver.rb [hunk 4]
```diff
@@ -37,31 +33,30 @@ def create
...
       Health::QualifyingActivity.new(
         source_type: @cha.class.name,
+        source_id: @cha.id,
         user_id: @user.id,
         user_full_name: @user.name_with_email,
-        date_of_activity: Date.current,
+        date_of_activity: @cha.completed_at.to_date,
         activity: :cha,
-        follow_up: 'Implement Comprehensive Health Assessment',
+        follow_up: 'This writer completed CHA and SSM with patient.',
```
Comment (eanders) [inline app/models/health/cha_saver.rb:57]: This seems odd, the SSM could have bee done days before the CHA, and what happens if they don't occur in the right order?

File: app/models/health/ssm_saver.rb [hunk 5]
```diff
@@ -23,23 +22,29 @@ def create
...
-            activity: :cha,
-            follow_up: 'Improve Patient Outcomes',
-            reached_client: :yes,
-            mode_of_contact: :in_person,
-            patient_id: @ssm.patient_id
-          )
+          @cha = @ssm.patient.recent_cha_form
+          if @cha.completed? && @cha.reviewed?
+            qualifying_activity = setup_qualifying_activity
+            qualifying_activity.save
```
Comment (eanders) [inline app/models/health/ssm_saver.rb:30]: This appears to be the inverse.  So maybe the comment is "Only add a QA after the CHA _and_ SSM have been completed.

File: app/views/health/release_forms/_form.haml [hunk 6]
```diff
@@ -44,4 +44,17 @@
 
     .col-6
       .form__actions--right
+        %button.btn.btn-secondary(aria-label="Cancel" pjax-modal-close type="button") Cancel
         = f.button :submit, 'Save', class: 'btn btn-primary'
+
+:javascript
+  $('#form_participation_signature_on').on('change', function(event) {
+    if ($('#form_signature_on').val() == '') {
```
Comment (eanders) [inline app/views/health/release_forms/_form.haml:52]: where is `#form_signature_on`?  Would it make more sense to do this in the controller?

**Review by eanders — COMMENTED** [review]
See inline notes.

File: app/models/health/careplan_saver.rb (hunk 2 above)
Comment (awisema) [reply app/models/health/careplan_saver.rb:82]: This is the pctp_signed QA, which already existed, and reflects that mode of contact is for the PCP, no changes in that were requested.

File: app/models/health/careplan_saver.rb (hunk 1 above)
Comment (awisema) [reply app/models/health/careplan_saver.rb]: Good catch! Fixed.

File: app/models/health/cha_saver.rb (hunk 3 above)
Comment (awisema) [reply app/models/health/cha_saver.rb:42]: This QA requires both the CHA and the SSM to have been completed, and can be done in either order, so both the ChaSaver and the SsmSaver need to generate the QA, just whichever is last.

File: app/models/health/ssm_saver.rb (hunk 5 above)
Comment (awisema) [reply app/models/health/ssm_saver.rb:30]: Good idea.

File: app/views/health/release_forms/_form.haml (hunk 6 above)
Comment (awisema) [reply app/views/health/release_forms/_form.haml:52]: These are the two date pickers at the top of the dialog, and this fills in the other one if one is filled in, and the other is blank. We could do this in the controller, but this seemed simpler than active cable or similar.

**Pull request merged**